scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/BucketMetadataUtil.java

```diff
@@ -95,19 +92,21 @@ public static PartitionMetadata create(
     int start = 0;
     while (start < total) {
       final List<ResourceId> input = directories.subList(start, Math.min(total, start + batchSize));
-      final List<Optional<BucketMetadata<K, V>>> result =
+      final List<BucketMetadata<K, V>> result =
           input
               .parallelStream()
-              .map(BucketMetadataUtil::<K, V>getMetadata)
+              .map(
+                  dir ->
+                      BucketMetadataUtil.getMetadata(dir)
+                          .map(metadata -> (BucketMetadata<K, V>) metadata)
+                          .orElseThrow(
+                              () ->
+                                  new RuntimeException(
+                                      "Could not find SMB metadata for source directory " + dir)))
```

Comment:
avoid extra allocation
```suggestion
                      (BucketMetadata<K, V>)
                          BucketMetadataUtil.getMetadata(dir)
                              .orElseThrow(
                                  () ->
                                      new RuntimeException(
                                          "Could not find SMB metadata for source directory "
                                              + dir)))
```

Reply:
good idea! thanks 👍